product_code/course_service/impl/src/main/scala/de/upb/cs/uc4/course/impl/CourseServiceImpl.scala

```diff
@@ -57,10 +57,18 @@ class CourseServiceImpl(
         .map { seq =>
           seq
             .filter { course =>
-              //If courseName query is set, we check that every whitespace seperated parameter is contained
-              courseName.isEmpty || courseName.get.toLowerCase.split("""\s+""").forall(course.courseName.toLowerCase.contains(_))
+              courseName match {
+                case None                => true
+                //If courseName query is set, we check that every whitespace separated parameter is contained
+                case Some(courseNameGet) => courseNameGet.toLowerCase.split("""\s+""").forall(course.courseName.toLowerCase.contains(_))
```

Comment:
How does this work?

Reply:
We have a sequence of courses and we filter that sequence in the following way: 

`courseName` is a string which contains whitespace separated tags which shall identify courses. For each `course` we check if every tag is included as a substring in the `course.courseName`. This allows some kind of fuzzy string search over the course names. For example if we have the tags ["online", "learning"] and the courses ["Online and Apdative Machine Learning", "Machine Learning 1"] we will return the filtered sequenced  ["Online and Apdative Machine Learning"]

Reply:
Okay thanks!